labgrid/external/remote_tmpdir.py

```diff
@@ -0,0 +1,53 @@
+"""
+    RemoteTmpdir is a class that maintains a temporary directory on a
+    remote linux system.
+    Ex. when copying a test script or binary to DUT, RemoteTmpdir assists
+    to prevent pollution of the target.
+
+    RemoteTmpdir is to be created in a fixture, which can take care of a
+    testmodule in a subdirectory. Recommended scope is function, to avoid
+    polution between tests.
+
+    See examples/remote_tmpdir
+"""
+
+import os
+
+
+class RemoteTmpdir:
+
+    # scriptdir is a directory relative to pytest root
+    def __init__(self, shell, testdir=None):
+        stdout, stderr, returncode = shell.run("mktemp -d")
+        assert returncode == 0
+        assert len(stdout) == 1
+        # Use path to prefix copied shells
+        # shell.run(tmp.path+"helloworld")
+        self.path = stdout[0] + '/'
+        self.shell = shell
```

Comment:
By storing `self.filetransfer`, there is no need for `self.shell`, as it is only used here in `__init__()`.